### PR TITLE
Enforce tinyint on Category fields.

### PR DESCRIPTION
**Pull request by beckyvb**
Fixes issue where Categories cannot be saved or edited with strict mode enabled.

### CODE REVIEW

File: applications/vanilla/controllers/class.settingscontroller.php [hunk 1]
```diff
@@ -565,6 +565,17 @@ public function editCategory($CategoryID = '') {
             }
             $this->Form->setFormValue('CustomPoints', (bool)$this->Form->getFormValue('CustomPoints'));
 
+            // Enforces tinyint values on boolean fields to comply with strict mode
+            $tinyintFields = array('HideAllDiscussions', 'Archived', 'AllowFileUploads');
```
Comment (linc) [inline applications/vanilla/controllers/class.settingscontroller.php]: all the cool kids use short arrays now

**Comment by linc** [comment]
What do you think about using the least-abstract version, like `CustomPoints` on the line prior?

 ```
$this->Form->setFormValue('HideAllDiscussions', forceBool($this->Form->getFormValue('HideAllDiscussions'), '0', '1', '0'));
$this->Form->setFormValue('Archived', forceBool($this->Form->getFormValue('Archived'), '0', '1', '0'));
$this->Form->setFormValue('AllowFileUploads', forceBool($this->Form->getFormValue('AllowFileUploads'), '0', '1', '0'));
```

**Comment by beckyvb** [comment]
Yeah, I agree with that.

**Comment by linc** [comment]
lol that commit message is the saddest

**Pull request merged**